src/pages/images/actions/BulkDeleteImageBtn.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { FC, useState } from "react";
+import { deleteImageBulk } from "api/images";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { ConfirmationButton, useNotify } from "@canonical/react-components";
+import { useEventQueue } from "context/eventQueue";
+import { getPromiseSettledCounts } from "util/helpers";
+import { pluralize } from "util/instanceBulkActions";
+
+interface Props {
+  fingerprints: string[];
+  project: string;
+  onStart: () => void;
+  onFinish: () => void;
+}
+
+const BulkDeleteImageBtn: FC<Props> = ({
+  fingerprints,
+  project,
+  onStart,
+  onFinish,
+}) => {
+  const eventQueue = useEventQueue();
+  const notify = useNotify();
+  const [isLoading, setLoading] = useState(false);
+  const queryClient = useQueryClient();
+
+  const count = fingerprints.length;
+
+  const handleDelete = () => {
+    setLoading(true);
+    onStart();
+    void deleteImageBulk(fingerprints, project, eventQueue).then((results) => {
+      const { fulfilledCount, rejectedCount } =
+        getPromiseSettledCounts(results);
+      if (fulfilledCount === count) {
+        notify.success(
```

Comment:
making a mental note here to implement floating notificaiton once this is merged